pkg/api/steve/supportconfigs/generator.go

```diff
@@ -24,33 +25,87 @@ import (
 
 const (
 	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
-	Endpoint            = "/v1/generateSUSERancherSupportConfig"
+	Endpoint = "/v1/generateSUSERancherSupportConfig"
+	// NOTE: the name of the configmap will be the same for both Bring-Your-Own (BOY) and
+	// Pay-As-You-Go (PAYG) offerings. However, they will be in different namespaces to avoid
+	// clashing.
 	cspAdapterConfigmap = "csp-config"
 	tarContentType      = "application/x-tar"
 	logPrefix           = "support-config-generator"
+	// Pay-As-You-Go offering
+	PAYG = "PAYG"
+	// Bring-Your-Own offering
+	BYO = "BYO"
 )
 
 var errNotFound = errors.New("not implemented")
 
+var (
+	configMaps   v1.ConfigMapInterface
+	cspNamespace string
+)
+
 // Handler implements http.Handler - and serves supportconfigs (tar file which contains support relevant information)
 type Handler struct {
-	ConfigMaps           v1.ConfigMapInterface
+	scaledContext        *config.ScaledContext
```

Comment:
This change (storing the configmap interface as a var, and storing the entire scaled context on the handler) seems unnecessary and more flawed than the original approach. 

Can you elaborate on why this change was made?

Reply:
I needed it to initialize the ConfigMaps, due to my lack of understanding of other options like the NamespaceAll option that you suggested later on. I will use that instead.